server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -607,20 +613,29 @@ public RestApiVersion getRestApiVersion() {
 
     public static class MediaTypeHeaderException extends RuntimeException {
 
-        private String failedHeaderName;
+        private String message;
+        private Set<String> failedHeaderNames;
+        private Object[] params;
+
+        MediaTypeHeaderException(Set<String> failedHeaderNames, String message, Object... params) {
+            this.params = params;
+            this.message = message;
+            this.failedHeaderNames = failedHeaderNames;
+        }
 
-        MediaTypeHeaderException(final IllegalArgumentException cause, String failedHeaderName) {
+        MediaTypeHeaderException(final RuntimeException cause, String... failedHeaderNames) {
             super(cause);
-            this.failedHeaderName = failedHeaderName;
+            this.failedHeaderNames = Set.of(failedHeaderNames);
+            this.message = "Invalid media-type value on header " + this.failedHeaderNames;
```

Comment:
Little wording/formatting nit -- maybe "on headers [HeaderA, HeaderB, HeaderC]"? What do you think?